### PR TITLE
feat(offcanvas): add static backdrop option

### PR DESCRIPTION
**Pull request by jnizet**
this aligns with Bootstrap 5.2.0 functionality

Before submitting a pull request, please make sure you have at least performed the following:

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/CONTRIBUTING.md) and [DEVELOPER.md](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/DEVELOPER.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated any applicable tests.
 - [x] added/updated any applicable API documentation.
 - [x] added/updated any applicable demos.

### CODE REVIEW

File: src/offcanvas/offcanvas-config.ts [hunk 1]
```diff
@@ -24,10 +24,11 @@ export interface NgbOffcanvasOptions {
 
   /**
    * If `true`, the backdrop element will be created for a given offcanvas.
+   * If 'static', clicking the backdrop won't close the offcanvas.
    *
    * Default value is `true`.
    */
-  backdrop?: boolean;
+  backdrop?: boolean | 'static';
```
Comment (maxokorokov) [inline src/offcanvas/offcanvas-config.ts:31]: You should type `backdrop: boolean | 'static' = true` in `NgbOffcanvasConfig` too (below in this file).
At the moment it is `backdrop = true` and typescript is unhappy when one tries to set it to 'static':

Screenshot 2022-08-02 at 16 01 26

**Review by maxokorokov — CHANGES_REQUESTED** [review]
Hey, @jnizet!
LGTM apart from small API typing issue in the `NgbOffcanvasConfig`

**Comment by jnizet** [comment]
Good catch again @maxokorokov. Fixed.

**Review by maxokorokov — APPROVED** [review]
LGTM, thanks!

**Pull request merged**